### DISCUSSION TITLE
Can i open and use mindjet mindmanager files .mmap in freeplane ?

### DISCUSSION THREAD

**Discussion by mulahnah** [freeplane/freeplane#1334] [Help]
Can i open and use mindjet mindmanager files .mmap in freeplane ?

**Comment by nnako** [comment]
Hi @mulahnah , sorry for the delay in answering your question...

As far as I see it, you can not directly use `.mmap` files in Freeplane. The `.mmap` file format is a binary (and proprietary) file format. Although it is a ZIP file, it contains a lot of binary content which can only be evaluated with the appripriate internal knowledge. That means that you need to do the following things to be able to read the map data from within another application (here: Freeplane):

- export the mindmap data into a text format (see https://onlinehelp.mindjet.com/help/MindManager/82/GER/save_a_map.htm), hoping that all the details are represented within the export format.
- identify the XML syntax used to represent the MindManager information
- parse the XML file structure (using e.g. the Python programming language and e.g. its [`xml`](https://docs.python.org/3/library/xml.etree.elementtree.html) library)
- create a `.mm` file which can natively be read into Freeplane. You could use the [`freeplane-python.io`](https://github.com/nnako/freeplane-python-io) library to do this.

sorry for this vague answer. I don't use MindManager and so I have never tried to analyze the native file structure. Anyway, hope you have an impression about the necessary efforts in order to convert `.mmap` files to `.mm` files. It would be a nice project to do. I think, there is no easy solution to your problem.

**Comment by piskernik** [comment]
Yes, you can use Mindmanager mind map files .mmap by importing them via File -> Branch import -> from MindManager X5 Mindmap... This works actually really well for standard branches including texts. If you were using an old Mindmanager Mac Version you have to update it first to a version above 10 (see here [Mindmanager Community](https://community.mindmanager.com/topic/1246-i-get-zip-archive-processing-error-when-opening-old-mmap-files)), otherwise the files can not be imported.
